src/stpipe/step.py

```diff
@@ -431,6 +463,42 @@ def run(self, *args):
             step_result = None
 
             self.log.info("Step %s running with args %s.", self.name, args)
+
+            # Check for explicit disable for CRDS parameters
+            disable_crds_steppars = kwargs.pop("disable_crds_steppars", None)
+
+            # Get parameters from user
+            parameters = None
+            if kwargs:
+                # Unset initialization for any provided override keywords
+                for key in kwargs:
+                    if key in self._initialized:
+                        self._initialized[key] = False
+                parameters = kwargs
+
+            # Get parameters from CRDS
+            if self._validate_kwds:
+                # Get the first filename, if available
+                filename = None
+                if len(args) > 0:
+                    filename = args[0]
+
+                # Build config from CRDS + user keywords
+                try:
+                    parameters, _ = self.build_config(
+                        filename, disable=disable_crds_steppars, **kwargs
+                    )
+                except (NotImplementedError, FileNotFoundError):
+                    # Catch steps that cannot build a config
+                    # (e.g. post hooks created from local functions,
+                    # missing input files)
+                    self.log.warning(f"Cannot retrieve CRDS keywords for {self.name}.")
```

Comment:
My vote is for making this an exception. If the expectation is that this will fetch CRDS parameters and that fails an error sees more useful than output that is incorrect because it doesn't use those parameters.

Reply:
Sure, I can make this an exception.  I originally made it a warning because things like post-hooks that aren't proper steps might make it to this clause, but I will make sure that case is separately handled.

Reply:
Done.  CRDS checks are now always disabled for pre and post hooks.